source/extensions/filters/network/http_connection_manager/config.cc

```diff
@@ -477,9 +483,11 @@ HttpConnectionManagerConfig::HttpConnectionManagerConfig(
     }
     if (!upgrade_config.filters().empty()) {
       std::unique_ptr<FilterFactoriesList> factories = std::make_unique<FilterFactoriesList>();
+      DependencyManager upgrade_dependency_manager;
       for (int32_t j = 0; j < upgrade_config.filters().size(); j++) {
-        processFilter(upgrade_config.filters(j), j, name, *factories, "http upgrade",
-                      j == upgrade_config.filters().size() - 1);
+        processFilter(upgrade_config.filters(j), j, name, "http upgrade",
+                      j == upgrade_config.filters().size() - 1, *factories,
+                      upgrade_dependency_manager);
```

Comment:
The upgrade filter chain is used instead of the regular filter chain when an upgrade happens (e.g. websocket), so I think we should be applying the same kind of validation to the upgrade filter chain as we do for the regular one? 

cc @alyssawilk

Reply:
Done.